clang/test/SemaSYCL/unique_stable_name.cpp

```diff
@@ -0,0 +1,223 @@
+// RUN: %clang_cc1 %s --std=c++17 -fsyntax-only -fsycl-is-device -Wno-unused -verify
+
+template <typename KernelName, typename KernelType>
+[[clang::sycl_kernel]] void kernel_single_task(KernelType kernelFunc) { // #kernelSingleTask
+  kernelFunc();
+}
+
+// kernel3 - expect error
+// The current function is named with a lambda (i.e., takes a lambda as a
+// template parameter. Call the builtin on the current function
+// Then it is passed to kernel
+// Error is thrown because the current function is passed to the kernel,
+// thus part of the latter's name, after its unique stable is already computed
+// Current function is a part of that name, causing its unique_stable_name
+// mangling to change
+template <typename Func>
+void kernel3func(const Func &F3) {
+  // expected-error@#kernelSingleTask{{kernel instantiation changes the result of an evaluated '__builtin_unique_stable_name'}}
```

Comment:
Where does this 2nd error come from?  it looks like it should be associated with some instantiation tree, so this is probably the wrong location.

Reply:
It comes directly from the test output (emitted without -verify). No other stack trace was emitted other than the error and the note below, associated with the builtin location. Not sure how to approach this.

Reply:
Note my previous comment in emails, the instantiation trace is only printed 1x, even if there are multiple diagnostics.  So this belongs with whatever initialization was 'above' this one.  LIkely with a comment.

Reply:
I would expect it to any time that a single kernel instantiation causes multiple diagnostics.